include/mpm_explicit.tcc

```diff
@@ -682,6 +682,20 @@ bool mpm::MPMExplicit<Tdim>::solve() {
   // Check point resume
   if (resume) this->checkpoint_resume();
 
+  // Create remove steps (excavation)
+  if (!particle_props["remove"].empty()) {
```

Comment:
We will add `remove` as a bool and unsigned `remove_step` inside the existing particle sets.